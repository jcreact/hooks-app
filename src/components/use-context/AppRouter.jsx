import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
} from 'react-router-dom';

// componentes globales.
import { NavBar } from './NavBar.jsx';

// Componentes.
import { HomePage } from './HomePage.jsx';
import { LoginPage } from './LoginPage.jsx';
import { AboutPage } from './AboutPage.jsx';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <main className="container-fluid">
                    <Switch>
                        <Route exact path="/" component={HomePage} />
                        <Route exact path="/about" component={AboutPage} />
                        <Route exact path="/login" component={LoginPage} />

                        <Redirect to="/" />
                    </Switch>
                </main>
            </div>
        </Router>
    );
};
