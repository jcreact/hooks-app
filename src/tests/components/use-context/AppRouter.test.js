
import { mount } from 'enzyme';
import React from 'react';

//Componente
import { AppRouter } from '../../../components/use-context/AppRouter.jsx';

//Hook
import { UserContext } from '../../../components/use-context/UserContext.js';

describe(`Pruebas en el componente 'HomePage.jsx'.`, () => {


    const user = {
        name: 'Jos√© Palma',
        email: 'test@correo.com'
    };

    const setUser = jest.fn();


    const app = mount(
        <UserContext.Provider
            value={{ user, setUser }}
        >
            <AppRouter />
        </UserContext.Provider>
    );

    test(`Debe renderizar el componente HomePage correctamente.`, () => {
        expect(app).toMatchSnapshot();
    });

});
