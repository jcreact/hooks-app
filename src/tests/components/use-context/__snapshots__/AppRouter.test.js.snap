// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente 'HomePage.jsx'. Debe renderizar el componente HomePage correctamente. 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <i
          className="bi bi-bug"
        />
         React App | useContext
      </a>
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <main
    className="container-fluid"
  >
    Array [
      <h1
        className="mt-3"
      >
        Home Page
      </h1>,
      <hr />,
      <pre
        className="font-monospace"
      >
        {
   "name": "Jos√© Palma",
   "email": "test@correo.com"
}
      </pre>,
    ]
  </main>
</div>
`;
